cdap-docs/deploy.sh

```diff
@@ -0,0 +1,201 @@
+#   3 scenarios (assuming remote base web directory=/var/www/html/cdap and VERSION=2.8.1)
+#     OPT_DIR is set via environment variables
+#         => remote directory=/var/www/html/cdap/${VERSION}-${OPT_DIR}
+#     OPT_DIR is not set AND branch=release/* or branch=develop/*
+#         => remote directory=/var/www/html/cdap/2.8.1
+#     OPT_DIR is not set AND branch=anything else (e.g. feature/*, hotfix/*, etc..)
+#         => remote directory=/var/www/html/cdap/${VERSION}-${BRANCH_NAME}
+set_remote_dir () {
+  DOC_DIR=${OPT_DIR:-${BRANCH_NAME}}
+  convert_branch_name
+  BRANCH=''
+  if [[ "${DOC_DIR}" == release* || "${DOC_DIR}" == develop* ]]; then
+    REMOTE_DIR=''
+  else
+    #REMOTE_DIR=branches/${DOC_DIR}
+    REMOTE_DIR=${VERSION}-${DOC_DIR}
+    BRANCH=yes
+  fi
+  decho "SUBDIR=${REMOTE_DIR}"
+}
+
+# parameters that can be passed to this script (as environment variables)
+DEBUG=${DEBUG:-no}
+DEPLOY_TO_STG=${DEPLOY_TO_STG:-no}
+DEPLOY_TO_DOCS=${DEPLOY_TO_DOCS:-no}
+DOCS_SERVER1=${DOCS_SERVER1:-docs1.cask.co}
+DOCS_SERVER2=${DOCS_SERVER2:-docs2.cask.co}
+STG_SERVER=${STG_SERVER:-docs-staging.cask.co}
+REMOTE_STG_BASE=${REMOTE_STG_BASE:-/var/www/html/${PROJECT}}
+REMOTE_DOCS_BASE=${REMOTE_DOCS_BASE:-/var/www/docs/${PROJECT}}
+# OPT_DIR (optional)
+
+decho "OPT_DIR=${OPT_DIR}"
+decho "BRANCH_NAME=${BRANCH_NAME}"
+
+get_version
+get_project
+set_remote_dir
+
+#
+USER=bamboo
+PROJECT_DOCS=${PROJECT}-docs
+ZIP_FILE=${PROJECT}-docs-${VERSION}-web.zip
+FILE_PATH=${BUILD_WORKING_DIR}/${PROJECT}/${PROJECT_DOCS}/build
+DOCS_SERVERS="${DOCS_SERVER1} ${DOCS_SERVER2}"
+REMOTE_STG_DIR="${REMOTE_STG_BASE}/${REMOTE_DIR}"
+REMOTE_DOCS_DIR="${REMOTE_DOCS_BASE}/${REMOTE_DIR}"
+
+RSYNC_OPTS='-aPh'
+SSH_OPTS='ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null'
+RSYNC_PATH='sudo rsync'
+
+die ( ) { echo ; echo "ERROR: ${*}" ; echo ; exit 1; }
+
+if [ "${PROJECT}" == '' ]; then
+  echo "PROJECT not defined"
+  exit 1
+fi
+
+################################################################################
+
+# copy zip file to local_dir's unzipped directory (e.g. cdap-docs/build/3.0.0-SNAPSHOT/)
+function copy_zip_file () {
+  local _zip_file=${1}
+  local _local_dir=${2}
+  local _version=${3}
+  decho "cp ${_local_dir}/${_zip_file} ${_local_dir}/${_version}/"
+  cp ${_local_dir}/${_zip_file} ${_local_dir}/${_version}/
+}
+
+# create remote directory prior to rsync
+function make_remote_dir () {
+  local _user=${1}
+  local _host=${2}
+  local _remote_dir=${3}
+  decho "ssh ${_user}@${_host} \"sudo mkdir -p ${_remote_dir}\""
+  ssh ${_user}@${_host} "sudo mkdir -p ${_remote_dir}" || die "could not create ${_remote_dir} directory on ${_host}"
+  decho ""
+}
+
+# rsync local dir to remote dir
+function sync_local_dir_to_remote_dir () {
+  local _user=${1}
+  local _host=${2}
+  local _remote_dir=${3}
+  local _zip_file=${4}
+  local _local_dir=${5}
+  local _version=${6}
+  if [ "${BRANCH}" == '' ]; then
+    _remote_dir=${_remote_dir}/${_version}
```

Comment:
seems like there is a bug here.  this function and make_remote_dir() above are called with the same _remote_dir, but it is modified here and not in make_remote_dir().  are we syncing to a directory not created yet?  also, isn't the remote_dir already set by set_remote_dir()? why is it modified again?